docs/fundamentals/code-analysis/style-rules/includes/dotnet-style-prefer-collection-expression.md

```diff
@@ -0,0 +1,15 @@
+---
+ms.topic: include
+---
+
+### dotnet_style_prefer_collection_expression
+
+| Property                 | Value                                     | Description                           |
+|--------------------------|-------------------------------------------|---------------------------------------|
+| **Option name**          | dotnet_style_prefer_collection_expression |                                       |
+| **Option values**        | `true` &#124; `when_types_exactly_match`  | Prefer to use collection expressions only when types match exactly, for example, `List<int> list = new List<int>() { 1, 2 };`. |
+|                          | `when_types_loosely_match`\*              | Prefer to use collection expressions even when types match loosely, for example, `IEnumerable<int> list = new List<int>() { 1, 2 };`. The targeted type must match the type on the right-hand side or be one of the following types: <xref:System.Collections.Generic.IEnumerable%601>, <xref:System.Collections.Generic.ICollection%601>, <xref:System.Collections.Generic.IList%601>, <xref:System.Collections.Generic.IReadOnlyCollection%601>, <xref:System.Collections.Generic.IReadOnlyList%601>. |
```

Comment:
@CyrusNajmabadi should take a look at these two lines. Currently, the collection expressions spec guarantees that `IList<T>` products a `List<T>` as the concrete type. That could mean that case should be in "when types exactly match", but I'm not sure the analyzer implemented it that way.

Reply:
>  That could mean that case should be in "when types exactly match", but I'm not sure the analyzer implemented it that way.



The analyzer is implemented that way.  When we see an `ICollection<T>` or `IList<T>` target, and we see the original type was `List<T>`, we will offer the change even if hte setting is "when types exactly match" as the final emitted type will exactly match the original type the user specified.